src/DataSet.cpp

```diff
@@ -245,7 +246,16 @@ int DataSet::readPsm(const std::string& line, const unsigned int lineNr,
     }
     proteins.swap(myPsm->proteinIds); // shrink to fit
   }
-  
+
+  if (label == -1) {
+    for(auto const& value: myPsm->proteinIds) { 
+    std::string token = value.substr(0, value.find("_") + 1);
```

Comment:
This might be a strong assumption, e.g. what if your prefix contains an underscore or if your prefix doesn't contain an underscore.

Reply:
I'm not sure how to solved this at the moment

Reply:
Fixed in 1030355